Sources/Plasma/CoreLib/hsStream.cpp

```diff
@@ -109,33 +115,32 @@ void hsStream::CopyToMem(void* mem)
 
 //////////////////////////////////////////////////////////////////////////////////
 
-uint32_t hsStream::WriteSafeStringLong(const ST::string &string)
+size_t hsStream::WriteSafeStringLong(const ST::string &string)
 {
-    uint32_t len = string.size();
-    WriteLE32(len);
+    size_t len = string.size();
+    WriteLE32(uint32_t(len));
```

Comment:
This may be overly pedantic, but...

Although it's extremely unlikely we'd ever have a string >2**32 characters long, this change would break it.  If we can only write a 32-bit value, we should ensure the written string length is limited to the same range, either with a 32-bit type (the previous `uint32_t`) or with an explicit range check (as is done below in `WriteSafeString`).